solarforecastarbiter/metrics/tests/test_probabilistic.py

```diff
@@ -0,0 +1,143 @@
+    (np.asarray([10, 10]), np.asarray([100, 100]), np.asarray([8, 8]), 0.0),
+    (np.asarray([10, 10]), np.asarray([100, 100]), np.asarray([10, 10]), 0.0),
+    (np.asarray([10, 10]), np.asarray([100, 100]), np.asarray([17, 17]), 1.0),
+    (np.asarray([10, 10]), np.asarray([100, 100]), np.asarray([2, 14]), 0.5),
+])
+def test_brier_score(fx, fx_prob, obs, value):
+    assert prob.brier_score(fx, fx_prob, obs) == value
+
+
+@pytest.mark.parametrize("fx,fx_prob,ref,ref_prob,obs,value", [
+    (7, 100, 5, 100, 8, 1.0 - 1.0 / 1.0),
+    (10, 50, 5, 100, 8, 1.0 - 0.25 / 1.0),
+])
+def test_brier_skill_score(fx, fx_prob, ref, ref_prob, obs, value):
+    assert prob.brier_skill_score(fx, fx_prob, ref, ref_prob, obs) == value
+
+
+@pytest.mark.parametrize("fx,fx_prob,obs,value", [
+    (
+        np.asarray([10, 10]),
+        np.asarray([100, 100]),
+        np.asarray([8, 8]),
+        0.0,
+    ),
+    (
+        np.asarray([3, 4]),
+        np.asarray([100, 100]),
+        np.asarray([10, 11]),
+        1.0,
+    ),
+    (
+        np.asarray([10, 5]),
+        np.asarray([100, 50]),
+        np.asarray([8, 8]),
+        0.125,
+    ),
+    (
+        np.asarray([10, 10]),
+        np.asarray([100, 50]),
+        np.asarray([8, 8]),
+        0.125,
+    ),
+])
+def test_reliability(fx, fx_prob, obs, value):
+    assert prob.reliability(fx, fx_prob, obs) == value
+
+
+@pytest.mark.parametrize("fx,fx_prob,obs,value", [
+    (
+        np.asarray([10, 10]),
+        np.asarray([100, 100]),
+        np.asarray([8, 8]),
+        0.0,
+    ),
+    (
+        np.asarray([8, 8]),
+        np.asarray([100, 100]),
+        np.asarray([10, 10]),
+        0.0,
+    ),
+    (
+        np.asarray([10, 5]),
+        np.asarray([100, 50]),
+        np.asarray([8, 8]),
+        0.25,
+    ),
+])
+def test_resolution(fx, fx_prob, obs, value):
+    assert prob.resolution(fx, fx_prob, obs) == value
+
+
+@pytest.mark.parametrize("fx,obs,value", [
+    # scalar inputs
+    (10, 8, 0.0),
+    (5, 8, 0.0),
+
+    # vector inputs
+    (np.asarray([10, 5]), np.asarray([8, 8]), 0.25),
+    (np.asarray([8, 8]), np.asarray([10, 10]), 0.0),
+])
+def test_unc(fx, obs, value):
+    assert prob.uncertainty(fx, obs) == value
+
+
+@pytest.mark.parametrize("fx,fx_prob,obs", [
+    # scalar inputs
+    (np.asarray([10]), np.asarray([100]), np.asarray([8])),
+    (np.asarray([10]), np.asarray([100]), np.asarray([10])),
+    (np.asarray([10]), np.asarray([100]), np.asarray([15])),
+
+    # vector inputs
+    (np.asarray([10, 10]), np.asarray([100, 100]), np.asarray([8, 8])),
+    (np.asarray([10, 5]), np.asarray([100, 50]), np.asarray([8, 8])),
+    (np.asarray([8, 8]), np.asarray([100, 100]), np.asarray([10, 10])),
+])
+def test_brier_decomposition(fx, fx_prob, obs):
+    bs = prob.brier_score(fx, fx_prob, obs)
+    rel = prob.reliability(fx, fx_prob, obs)
+    res = prob.resolution(fx, fx_prob, obs)
+    unc = prob.uncertainty(fx, obs)
+    assert bs == rel - res + unc
```

Comment:
this equality will fail for forecast probabilities that are rounded in the reliability and resolution calculation

Reply:
Good point. I'll see if adding `pytest.approx()` is sufficient for handling the effects of rounding the reliability and resolution calculations.

Reply:
Adding `pytest.approx()` seems to be sufficient for handling the effects of rounding in the reliability and resolution calculations.